stripzip.py

```diff
@@ -91,9 +103,10 @@ def purify_extra_data(mm, offset, length):
         # reset last_mod_date
         values[5] = 0x21
         local_file_header_struct.pack_into(mm, offset, *values)
-        offset += local_file_header_struct.size + compressed_size + name_length + extra_field_length
+        offset += local_file_header_struct.size + name_length
         if extra_field_length != 0:
-            purify_extra_data(mm, offset-extra_field_length-compressed_size, extra_field_length)
+            compressed_size = purify_extra_data(mm, offset, extra_field_length, compressed_size)
```

Comment:
would it make sense for `purify_extra_data` to return the offset to the end of whatever structure it is processing?

thinking of if, can the handing of the ZIP64 header be moved outside of `purify_extra_data`/does it make sense?

Reply:
1. I tend to prefer forward progressing only per function, without offset passing back and forth. That's the reason for drifting from the original code porting and removal of offset algebra, when possible.
2. yup, `purify_extra_data` does not represent the correctly what is happening in that function. It's more an `handle_extra_data` function atm.